### DISCUSSION TITLE
How to assign tags to scripts via API

### DISCUSSION THREAD

**Discussion by lordarokh** [microsoft/testbase#36] [Announcements]
We are creating Test Base Application package via the API's.
We have created a package aligned with the Sample examples that are provided.
But the package validation failed. When we check the reason, we couldn't find any explanation.
As of troubleshooting, we edited our package to see if anything which might be missing.
And there we encountered the message that, "TAGS" are missing for the scripts. 
For testing purposes, we have manually assigned the "TAGS" to the scripts. 
However, when we compare the JSON data with the one which we assigned tags manually against our failed package JSON data, we do not see any difference.

So we are lost on how the TAGS are assigned to the scripts, if we want to work thru API's.

Any ideas?

Thanks in advance.

**Comment by Tinacyt** [comment]
Hi there, 

Thank you for starting onboarding your package to Test Base via API! 😊

In order to better support your 'validation failure' case, may I know which test type you are setting for the package? Out of Box testing or Functional testing?

Best,
Tina

**Reply by lordarokh** [reply]
Currently we use only "**Out of Box Test**" scenario.
We realized that on some of our test package the TAGS are properly assigned. But on some packages, they are not. When we compare 2 packages where there is no TAG assignment with a package against a package with TAG assignment, we do not see any difference in JSON data information. The NAME and ACTION properties are properly populated with correct values.

Also, the documentation "[Packages - Create](https://learn.microsoft.com/en-us/rest/api/testbase/packages/create?tabs=HTTP)" provides the sample JSON for testing section as follows (we only present one action here) :
"commands": [
          {
            "name": "Install",
            "action": "Install",
            "contentType": "Path",
            "content": "app/scripts/install/job.ps1",
            "runElevated": true,
            "restartAfter": true,
            "maxRunTime": 1800,
            "runAsInteractive": true,
            "alwaysRun": true,
            "applyUpdateBefore": false
          },
The property value of "NAME" is given as "Install". However if you perform the package-create API call, then it fails with following error:

{"error":{"code":"BadRequest","message":"Provided OutOfBoxTest Test command names [Install, Launch, Close, Uninstall] are not valid, supported command names [install, launch, close, 
uninstall]"}}

So, looking at the message, the package-create API is case sensitive for Name Property.

However, it is not clear how script tags assigned thru API calls.

Kind regards,

**Reply by yazeng** [reply]
@lordarokh "name" should be in lower case, "install". Action should be "Install". This is how we show script tags in API.
We'll update the example in "[Packages - Create](https://learn.microsoft.com/en-us/rest/api/testbase/packages/create?tabs=HTTP)".
